core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/manager/DataPlaneManagerImpl.java

```diff
@@ -152,16 +152,37 @@ private void processDataFlowRequest(DataFlowRequest request) {
             store.completed(request.getProcessId());
         } else {
             transferService.transfer(request).whenComplete((result, exception) -> {
+
                 if (request.isTrackable()) {
                     // TODO persist TransferResult or error details
                     store.completed(request.getProcessId());
                 }
+
+                if (transferProcessClient != null) {
+                    onTransferFinished(request, result, exception);
+                }
+
             });
         }
     }
 
+    private void onTransferFinished(DataFlowRequest request, StatusResult<Void> result, Throwable exception) {
+        if (exception != null) {
+            transferProcessClient.failed(request, exception.getMessage());
+        } else if (result.succeeded()) {
+            transferProcessClient.completed(request);
+        } else {
+            transferProcessClient.failed(request, result.getFailureDetail());
+        }
+    }
+
     public static class Builder {
-        private DataPlaneManagerImpl manager;
+        private final DataPlaneManagerImpl manager;
+
+        private Builder() {
+            manager = new DataPlaneManagerImpl();
```

Comment:
unrelated, but the CTor of the DataPlaneManagerImpl could be made `private`, to force the use of the Builder